src/main/java/frc/robot/swerve/SwerveDrive.java

```diff
@@ -0,0 +1,275 @@
+
+        ChassisSpeeds fieldOrientedChassisSpeeds = ChassisSpeeds.fromFieldRelativeSpeeds(
+                chassisSpeeds.vxMetersPerSecond,
+                chassisSpeeds.vyMetersPerSecond,
+                chassisSpeeds.omegaRadiansPerSecond,
+                new Rotation2d(getYaw())
+        );
+
+        if (chassisSpeeds.equals(new ChassisSpeeds(0, 0, 0))) {
+            for (SwerveModule module : modules) {
+                module.neutralOutput();
+            }
+        }
+
+        if (fieldOriented) {
+            setModuleStates(kinematics.toSwerveModuleStates(fieldOrientedChassisSpeeds));
+        } else {
+            setModuleStates(kinematics.toSwerveModuleStates(chassisSpeeds));
+        }
+    }
+
+    /**
+     * Sets the desired percentage of x, y and omega speeds for the frc.robot.swerve
+     *
+     * @param xOutput     percentage of the x speed
+     * @param yOutput     percentage of the y speed
+     * @param omegaOutput percentage of the omega speed
+     */
+    public void drive(double xOutput, double yOutput, double omegaOutput, boolean fieldOriented) {
+        ChassisSpeeds chassisSpeeds = new ChassisSpeeds(
+                SwerveConstants.MAX_X_Y_VELOCITY * xOutput,
+                SwerveConstants.MAX_X_Y_VELOCITY * yOutput,
+                SwerveConstants.MAX_OMEGA_VELOCITY * omegaOutput); //removed angleFF
+
+        drive(chassisSpeeds, fieldOriented);
+    }
+
+    public void periodic() {
+        updateModulePositions();
+        odometry.update(new Rotation2d(getYaw()), modulePositions);
+
+        loggerInputs.botPose[0] = odometry.getPoseMeters().getX();
+        loggerInputs.botPose[1] = odometry.getPoseMeters().getY();
+        loggerInputs.botPose[2] = odometry.getPoseMeters().getRotation().getRadians();
+
+        for (int i = 0; i < modules.length; i++) {
+            currentModuleStates[i] = modules[i].getModuleState();
+            loggerInputs.absolutePositions[i] = modules[i].getPosition();
+        }
+
+        Logger.recordOutput("SwerveDrive/currentModuleSates", currentModuleStates);
+
+        for (int i = 0; i < 3; i++) {
+            loggerInputs.currentSpeeds[i] =
+                    Utils.chassisSpeedsToArray(
+                            kinematics.toChassisSpeeds(
+                                    currentModuleStates[0],
+                                    currentModuleStates[1],
+                                    currentModuleStates[2],
+                                    currentModuleStates[3]
+                            ))[i];
+        }
+
+        loggerInputs.linearVelocity = Math.hypot(loggerInputs.currentSpeeds[0], loggerInputs.currentSpeeds[1]);
+
+        acceleration.update(loggerInputs.linearVelocity);
+        loggerInputs.acceleration = accelFilter.calculate(acceleration.get());
+
+        loggerInputs.supplyCurrent =
+                modules[0].getSupplyCurrent() + modules[1].getSupplyCurrent() + modules[2].getSupplyCurrent() + modules[3].getStatorCurrent();
```

Comment:
```suggestion
        double totalSupplyCurrent = 0;
        for (SwerveModule module : modules) {
            totalSupplyCurrent += module.getSupplyCurrent();
        }
        loggerInputs.supplyCurrent = totalSupplyCurrent;
```
Or:
```suggestion
        loggerInputs.supplyCurrent = Arrays.stream(modules)
                                    .mapToDouble(SwerveModule::getSupplyCurrent)
                                    .sum();
```
Or:
```suggestion
Arrays.stream(modules)
      .forEach(module -> loggerInputs.supplyCurrent += module.getSupplyCurrent());
```
Same for stator current.

Reply:
These are probably not needed because AdvantageKit can logs currents by itself.

Reply:
454d794d8a787736f42e5e0f24268845c8bcace8

Reply:
This is good, but are you sure logging these fields is necessary? Remember, all PDH data is already logged by AdvantageKit.

Reply:
But doesn't it log the port on the pdh? We don't really think about those.

Reply:
Indeed, it doesn't log motors' names, but in the rare case you want to see the electrical data you can quickly check what PDH port corresponds to the relevant motor.